cppcheck: Fix --with-gui option

aharrison24 opened the pull request:
- In 1.68, gui.qrc no longer references a cfg dir, so removed the
  previously-required inreplace.
- The QT build now requires HAVE_RULES to be explicitly enabled, so
  added that to the qmake invocation.
- Fixed a clang compile error caused by ambiguous closing template angle
  brackets.  This fix is already in upstream’s master branch.
- Made sure QT project builds in C++11 mode when using clang.  This fix
  is taken from upstream’s master branch.
- Caused .app to be installed directly in the prefix dir, not the bin
  dir - that was stopping ‘brew linkapps’ from working.

jacknagel commented:
Has the patch been sent upstream?

mikemcquaid commented on Library/Formula/cppcheck.rb:
```diff
@@ -19,6 +19,8 @@ class Cppcheck < Formula
   depends_on "pcre" if build.with? "rules"
   depends_on "qt" if build.with? "gui"
 
+  patch :DATA
```
Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

aharrison24 commented:
Both of these issues have been addressed in upstream's master branch, but have not yet made it into an official release of cppcheck.

I've modified the clang 'c++11 flag' part of the patch to exactly match what upstream have.  I've also added a comment to the patch line explaining that the patches have come from upstream.  I hope that gives enough information now.  Do you need relevant commit IDs from the upstream repo?

mikemcquaid commented on Library/Formula/cppcheck.rb:
```diff
@@ -19,6 +19,10 @@ class Cppcheck < Formula
   depends_on "qt" if build.with? "gui"
 
+  # Patches from upstream's master branch that are required for clang++
+  # builds.  These have not yet made it into an official release.
```
Please link to the upstream issue/commits, thanks!

mikemcquaid commented on Library/Formula/cppcheck.rb:
```diff
@@ -36,18 +40,14 @@ def install
           system "qmake", "HAVE_RULES=no"
         end
 
         system "make"
```
Is there no `make install` we can use?

aharrison24 commented:
OK, I've modified the formula to directly apply the appropriate diff from the upstream repository.

The 'install' target for the qmake build appears to do nothing, so I believe it is still necessary to copy the app bundle manually.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!
